### PR TITLE
Fixed handling of flags in Args

### PR DESCRIPTION
**Pull request by lutovich**
Currently after parsing `-option1=value -flag1 false -option2=value -flag2 false` both `false` would be recognised as orphan arguments.
This happens because parser does not consume any values after flags.
So `-flag=true` results in `flag-->true`, `-flag=false` results in `flag-->false`, `-flag` results in `flag-->true` and `-flag false` results in `flag-->true` & orphan `false`.

This PR fixes parsing and makes Args compatible with previous versions (<= 2.1).
When value that follows flag could be consumed as boolean it is consumed, otherwise it is treated as orphan.
PR also adds some tests for `BackupTool`.

### CODE REVIEW

File: enterprise/backup/src/test/java/org/neo4j/backup/BackupToolCmdArgumentsAcceptanceTest.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    private static final String HOST = "localhost";
+    private static final int PORT = 9090;
+    private static final String PATH = "/var/backup/neo4j/";
+
+    @Parameter( 0 )
+    public String[] args;
+    @Parameter( 1 )
+    public boolean expectedVerifyStoreValue;
+
+    @Parameters
```
Comment (davidegrohmann) [inline enterprise/backup/src/test/java/org/neo4j/backup/BackupToolCmdArgumentsAcceptanceTest.java]: you should add the parameter `name` to `@Parameters`

File: enterprise/backup/src/test/java/org/neo4j/backup/BackupToolCmdArgumentsAcceptanceTest.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    private static final String HOST = "localhost";
+    private static final int PORT = 9090;
+    private static final String PATH = "/var/backup/neo4j/";
+
+    @Parameter( 0 )
+    public String argsAsString;
+    @Parameter( 1 )
+    public boolean expectedVerifyStoreValue;
+
+    @Parameters( name = "args=({0})" )
```
Comment (lutovich) [inline enterprise/backup/src/test/java/org/neo4j/backup/BackupToolCmdArgumentsAcceptanceTest.java]: @davidegrohmann turned out there exist no format for printing 2D arrays via `MessageFormat`. that is why i've made first parameter `String` where all args are just joined with space as separator. Test method splits this string in line 88. Does this look sane?

**Comment by benbc** [comment]
@lutovich The PR comment says "makes Args compatible with previous versions (< 2.1)". That suggests that the bug was introduced in 2.1; is that correct? Or should that read "<=2.1"?

**Comment by lutovich** [comment]
@benbc Yeah, it should be '<='. Will fix description.

File: enterprise/backup/src/test/java/org/neo4j/backup/BackupToolCmdArgumentsAcceptanceTest.java (hunk 2 above)
Comment (davidegrohmann) [inline enterprise/backup/src/test/java/org/neo4j/backup/BackupToolCmdArgumentsAcceptanceTest.java]: I guess it is good enough.

**Comment by davidegrohmann** [comment]
@lutovich perhaps part of the description should be part of the first commit message?

**Comment by lutovich** [comment]
@davidegrohmann message improved

**Pull request merged**